src/lib.rs

```diff
@@ -54,6 +54,19 @@ fn ensure_current_version<'de, D: Deserializer<'de>>(d: D) -> Result<String, D::
     Ok(unchecked)
 }
 
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Default)]
+pub struct BuildInfo {
+    /// The hash of the contract WASM code.
+    pub wasm_hash: String,
```

Comment:
Unsure how we're to represent this, we should probably just make it an `Option`, in the case that the ABI is embedded, otherwise this would be cyclic.

Reply:
interesting, that could be valid. I was under the assumption that build info in general would be omitted when embedding, but that is valid that one might want the other build info in the embedded ABI metadata

Reply:
Yes, my intuition was for the build info to be omitted entirely when embedding, but I guess an argument could be made that someone would want some of this info inside wasm.

Reply:
Moved `wasm_hash` to metadata root, so build info can be embedded now if we wanted to.